src/python/grpcio/grpc/_auth.py

```diff
@@ -60,15 +69,7 @@ def __call__(self, context, callback):
         else:
             future = self._pool.submit(self._credentials.get_access_token)
         future.add_done_callback(
-            lambda x: self._get_token_callback(callback, x))
-
-    def _get_token_callback(self, callback, future):
-        try:
-            access_token = future.result().access_token
-        except Exception as e:
-            _sign_request(callback, None, e)
-        else:
-            _sign_request(callback, access_token, None)
+            lambda ignored_future: _get_token_callback(callback, future))
```

Comment:
Summary of our offline discussion: using the argument as opposed to closure will prevent future bugs that might arise by accidentally reassigning the `future` variable in a possible future CL (however improbable due to our pervasive use of single assignments where possible).